src/Microsoft.ML.SearchSpace/SearchSpace/SearchSpace{T}.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.ML.SearchSpace
+{
+    public class SearchSpace<T> : SearchSpace
+        where T : class, new()
```

Comment:
Move this class into the same file as the non-generic `SearchSpace` class.

Reply:
Hmmm I prefer having one class only in one file,, Is having generic and it's non-generic class in samee file the coding rule in mlnet?

Reply:
Its not a rule to my knowledge, but its been the style that has been done everywhere as far as I know. Let me double check. [Here](https://github.com/dotnet/machinelearning/blob/0577957256c296fdea2deb6b6e00e7be9b458167/src/Microsoft.Data.Analysis/GroupBy.cs) and [here](https://github.com/dotnet/machinelearning/blob/d849ba4c7831586018d5b2cca6bdd2b1fa228ddf/src/Microsoft.ML.AutoML/API/RunDetails/RunDetail.cs#L19) are good examples. While we haven't _always_ followed this rule, if the classes are short and if they have the same name (just 1 is generic and the other isn't) we pretty much have always put them in the same file. 

So I think it would be better to keep that style if possible and have these in the same file since this fits those conditions.

Reply:
### Resolved